flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -357,11 +359,26 @@ public void resumeConsumption() throws IOException {
         checkState(!isReleased.get(), "Channel released.");
         checkPartitionRequestQueueInitialized();
 
+        if (initialCredit == 0) {
+            unannouncedCredit.set(0);
+        }
+
         // notifies the producer that this channel is ready to
         // unblock from checkpoint and resume data consumption
         partitionRequestClient.resumeConsumption(this);
     }
 
+    private void onBlockingUpstream() {
+        if (initialCredit == 0) {
```

Comment:
Why can not we do the same thing for any number of credits not only for `initialCredit == 0`? Or does it slow down the load after resuming?

Reply:
We can also release the floating buffers if the initialCredit != 0. I think it does not slow down the load after resuming.